// Copyright 2021 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package google.internal.federated.plan;

option java_package = "com.google.internal.federated.plan";
option java_multiple_files = true;

// Provides the information needed to determine eligibility for tasks in a
// population.
message PopulationEligibilitySpec {
  // Eligibility-related information about each task in the population.
  repeated TaskInfo task_info = 1;

  message TaskInfo {
    // The name of the task.
    string task_name = 1;

    // The TaskAssignments method to use for the task.
    TaskAssignmentMode task_assignment_mode = 2;

    enum TaskAssignmentMode {
      TASK_ASSIGNMENT_MODE_UNSPECIFIED = 0;
      // Task assignment uses /TaskAssignments.StartTaskAssignment.
      TASK_ASSIGNMENT_MODE_SINGLE = 1;
      // Task assignment uses /TaskAssignments.PerformMultipleTaskAssignments.
      TASK_ASSIGNMENT_MODE_MULTIPLE = 2;
    }
  }
}
